Amend grammar to accept Elm 0.17 module declaration

ThomasWeiser opened the pull request:
Grammar now accepts both syntax variations:

Elm 0.16: `module Queue (..) where`
Elm 0.17: `module Queue exposing (..)`

eeue56 commented:
@deadfoxygrandpa ping

esad commented:
Would adding support for "port module" and "effects module" be too much effort?

ThomasWeiser commented:
Added a commit with:
- Support for special `port` and `effect` module syntax.
- Changed both `Elm.YAML-tmLanguage` and `Elm.tmLanguage`

Example of valid module declarations in 0.17:

``` elm
effect module WebSocket where { command = MyCmd, subscription = MySub } exposing (
    send
  , listen
  , keepAlive
  )
```

``` elm
port module Spelling exposing (..)
```

``` elm
module WebSocket.LowLevel exposing ( WebSocket
  , open, Settings
  , send, close, closeWith
  , bytesQueued
  , BadOpen(..), BadClose(..), BadSend(..)
  )
```

``` elm
module Main 
```

0.16:

``` elm
module Config ( title ) where
```

(Please note, that at the time of this writing GitHub highlights the code blocks above according to 0.16 rules. This repo [is the source](https://github.com/github/linguist/tree/master/vendor/grammars) of GitHub's Elm rules. So merging this PR should change also the highlighting here.)

glenjamin commented:
In case anyone wants to quickly apply this manually, you can try the following - which worked for me on OS X. You'll need to adjust the paths as appropriate

``` sh
cd /Users/glen/Library/Application\ Support/Sublime\ Text\ 3/Packages/Elm\ Language\ Support
curl https://github.com/deadfoxygrandpa/Elm.tmLanguage/pull/101.patch -L | patch -p1
```

igregson commented:
For @glenjamin's approach on Ubuntu, the directory to run the curl command from (at least for my specific st3 install):

```
~/.config/sublime-text-3/Packages/Elm\ Language\ Support
```

rtfeldman commented:
@deadfoxygrandpa this is a huge community pain point, and Sublime is now the only editor not updated for 0.17's breaking syntax changes...are you still up for maintaining this repo?

esad commented:
@deadfoxygrandpa Please consider moving this project to elm-community organization. What do you think @rtfeldman, would having support for major editors under elm-community make sense?

eeue56 commented:
This has already been moved to elm-community. Deadfoxygrandpa has been
unresponsive for about a month now

On Monday, May 30, 2016, Esad Hajdarevic notifications@github.com wrote:

> @deadfoxygrandpa https://github.com/deadfoxygrandpa Please consider
> moving this project to elm-community organization. What do you think
> @rtfeldman https://github.com/rtfeldman, would having support for major
> editors under elm-community make sense?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/deadfoxygrandpa/Elm.tmLanguage/pull/101#issuecomment-222474749,
> or mute the thread
> https://github.com/notifications/unsubscribe/ABFh_hNyPay6QXnO5qawdyJbFVwSdsmhks5qGs1MgaJpZM4IUQr9
> .

## 

Cheers,
Noah

noah@astraldynamics.co.uk
Astral Dynamics Cyf
Festri Cefn Y Waen, Deiniolen, Caernarfon, Gwynedd LL55 3NB
Company number 07726862
Registered in England and Wales

esad commented:
@eeue56 Should we make PR to https://github.com/wbond/package_control_channel to have the default package switch to the new repo?

ThomasWeiser commented:
Just opened a [PR](https://github.com/wbond/package_control_channel/pull/5551) to adopt the package by elm-community.

rtfeldman commented:
@ThomasWeiser probably makes sense to do the same for linguist, to update GH, yeah?

ThomasWeiser commented:
Opened [PR](https://github.com/github/linguist/pull/3027) at github/linguist.

ThomasWeiser commented:
Status report:
- Code rendering on GitHub is [updated](https://github.com/github/linguist/pull/3033).
- Sublime package: The Package Control maintainers [will wait](https://github.com/wbond/package_control_channel/pull/5551#issuecomment-223156934) 14 days for deadfoxygrandpa to come up on this issue.
